source/client/options_impl.cc

```diff
@@ -189,6 +189,17 @@ OptionsImpl::OptionsImpl(int argc, const char* const* argv) {
       "Add uniformly distributed absolute request-release timing jitter. For example, to add 10 us "
       "of jitter, specify .00001s. Default is empty / no uniform jitter.",
       false, "", "duration", cmd);
+  TCLAP::MultiArg<std::string> backend_endpoints(
+      "", "backend-endpoint",
+      "Backend endpoint overrides. "
+      "This argument is intended to be specified multiple times. "
+      "Nighthawk will generate the same traffic as usual, but "
+      "will spread it across all backend endpoints with "
+      "round robin distribution. "
+      "Endpoint formats: IPv4:port, [IPv6]:port, DNS:port. "
```

Comment:
I haven't really thought this through yet, but one fleeting thought I just had is if maybe it would make sense to make this interact with the rate limiters. Consider ramping vs round robin over multiple backends.. do we want to the ramp specific per backend? In that case, this could act like a multiplier of sorts, Nighthawk strifes to keep release timings close-to-equal across backends, e.g. one rate limiter aquisition causes a request to be released for each backend. Possibly this is the opposite of what's desirable, just a braindump.

Reply:
I rewrote it all so that we don't do anything hacky with the URI. Instead for multi-target mode the URI is omitted and  the path and endpoints are directly spelled out. I left the port mandatory in each endpoint since it would take extra logic to determine a default.

Reply:
About the rate limiter, I think I see what you mean. I think we might be all right as-is. If someone activates the multi-backend feature with 3 backends, they have signed up for a 3x reduction in the steady state rps seen at each backend. It sounds like Nighthawk today will also naturally produce a 3x reduction in the ramp rate seen at each backend. 

We don't have any ramping needs at the moment, and it would certainly be safe for us to accept any slow ramping.

Could the user who sets up 3 backends deliberately configure a 3x higher ramp rate to compensate for the 3x reduction that will occur?

Someone who needed to fully control the ramp rate on each backend could also run multiple Nighthawks as today.

By the way, we don't have that much confidence that this multiple backends mode will work better than running multiple Nighthawk processes. The sole motivation is to simplify our testbed by reducing the number of processes. The simplification could come into play more when we implement statistical models in the request source.

Reply:
sounds great; this makes sense to me.